src/components/home/index.js

```diff
@@ -42,15 +28,15 @@ class Gallery extends Component {
 
   render () {
     const { searchSummary, ...tileData } = mockData;
-    const { addMessageVisible } = this.state;
+    const { addMessageVisible, showAddMessage, hideAddMessage, yesFilter, noFilter, filterVisible } = this.props;
     return (
       <div className='homeContainer'>
         <Modal showModal={this.state.modalVisible} close={this.closeModal}/>
         <SearchBar />
         <SearchSummary {...searchSummary} />
         <Tags />
-        <Grid showAddMessage={this.showAddMessage} tileData={tileData}/>
-        {addMessageVisible && <AddMessage hideAddMessage={this.hideAddMessage} suggestedLocations='Croatia and Greece'/>}
+        <Grid yesFilter={yesFilter} noFilter={noFilter} filterVisible={filterVisible} showAddMessage={showAddMessage} tileData={tileData}/>
```

Comment:
You could give grid a container and make it a smart component to avoid passing the actions down.